clang/lib/CIR/CodeGen/CIRGenBuiltin.cpp

```diff
@@ -302,6 +302,678 @@ RValue CIRGenFunction::buildRotate(const CallExpr *E, bool IsRotateRight) {
   return RValue::get(r);
 }
 
+/// Get builtin name for unhandled BuiltinID in the function
+/// `CIRGenFunction::buildBuiltinExpr`. Thoese builtin names are handled
+/// in OG code `CodeGenFunction::EmitBuiltinExpr`.
+/// NOTE: Every time we implement handling of a case in
+/// the function `CIRGenFunction::buildBuiltinExpr`, it should be remove here.
+static std::string getUnhandledBuiltinName(unsigned builtinID) {
+  switch (builtinID) {
+  default:
+    return std::string();
+  case Builtin::BI__builtin_acos:
```

Comment:
These builtins are defined in tablegen, and I usually search in my build directory to find the corresponding `.inc` file (`tools/clang/include/clang/Basic/Builtins.inc` in this case). Those files define all the builtins using macros, and you can provide an appropriate macro definition before including the file to autogenerate this switch instead of needing to manually write out every intrinsic.

Reply:
Thanks for great advice. This situation is a bit different though because these builtin names are a selection of tools/clang/include/clang/Basic/Builtins.inc. They are actually a manual selection of builtins that has implementation in OG but not yet in CIRGen, And the idea was just to capture names and NYI early of those builtins (and them only) as those implemented in CIRGen should never hit the NYI I created here, and those not implemented in OG should go through and never hit NYI I created.
 i.e. I used command like 

> "sed -n '2573,6333p' /data/users/guojinhe/clangir-llvm-ghehg/clang/lib/CodeGen/CGBuiltin.cpp | grep -o 'Builtin::BI__[^:]*' | sed 's/^/case /;s/$/:/' "  to get them from oG. 
> and exclude what's from sed -n '305,1416p' /data/users/guojinhe/clangir-llvm-ghehg/clang/lib/CIR/CodeGen/CIRGenBuiltin.cpp  | grep "case Builtin:"
to pick them.

Yeah, let me know if we have better way to generate them

Reply:
With the current implementation, I think it'd be okay to have this function handle all built-ins and not just unimplemented ones. Bruno's suggestion changes that, of course.